src/analysis.ts

```diff
@@ -0,0 +1,13 @@
+import {DocumentDescriptor} from './ast/ast';
+import {SerializedAnalysis} from './serialized-analysis';
+
+export class Analysis {
+  private descriptors_: DocumentDescriptor[];
+  constructor(descriptors: DocumentDescriptor[]) {
+    this.descriptors_ = descriptors;
+  }
+
+  serialize(): SerializedAnalysis {
```

Comment:
Ultimately, I think this should be a function that takes an Analysis, not a method.